src/core/MRMedia.js

```diff
@@ -0,0 +1,242 @@
+import * as THREE from 'three';
+
+import { MRDivEntity } from 'mrjs/core/MRDivEntity';
+
+import { mrjsUtils } from 'mrjs';
+
+/**
+ * @class MRMedia
+ * @classdesc Base html media entity represented in 3D space. `mr-media`
+ * @augments MRDivEntity
+ */
+export class MRMedia extends MRDivEntity {
+    /**
+     * @class
+     * @description Constructs a base media entity using a UIPlane and other 3D elements as necessary.
+     */
+    constructor() {
+        super();
+        this.attachShadow({ mode: 'open' });
+
+        // Create the object3D. Dont need default value for geometry
+        // until the connected call since this will get overwritten anyways.
+        let material = new THREE.MeshStandardMaterial({
+            side: 0,
+            transparent: true,
+        });
+        // Object3D for MRMedia (mrimage,mrvideo,etc) is the actual image/video/etc itself in 3D space
+        this.object3D = new THREE.Mesh(undefined, material);
+        this.object3D.receiveShadow = true;
+        this.object3D.renderOrder = 3;
```

Comment:
Not a review comment but just a question for existing code so not blocking: What is this `renderOrder = 3` for? Adding a comment would be helpful to users.

Reply:
right now my understanding for mrjs is we're using renderOrder kind of like an addition to Z-depth to get around some of the overhead of that, and then i think we were going to switch it to an enum to make it cleaner-code so a comment wouldnt be necessary

i'm not fully up to speed on it otherwise though, but @michaelthatsit would have a better answer to add here if i missed anything in my above explanation

Reply:
i think this older branch might've also been related to it: `349-integrate-depth-sensing-to-replace-render-order-hack`

Reply:
@michaelthatsit I'm happy if you explain because as I explained at other places I'm thinking of using `renderOrder` for masking (stencil) system.

Anyways, this is just a question, not a comment for any blocking issue.

Reply:
marking this as resolved just to make sure i can keep an eye on remaining todos for this pr

opened up a [separate issue](https://github.com/Volumetrics-io/mrjs/issues/480) so @michaelthatsit can follow up with that discussion for you @takahirox :)